### PR TITLE
Fix Savce issues in wifi_manager.c: strcpy=>strncpy, specifying the l…

### PR DESCRIPTION
**Pull request by davidfather**
…ength of string includig '\0'.

1. replace strcpy with strncpy
2. specify the length of string including '\0'.

### CODE REVIEW

File: framework/src/wifi_manager/wifi_manager.c [hunk 1]
```diff
@@ -186,7 +186,7 @@ static void wifi_linkdown_event_func(void)
 	if (g_manager_info.mode == STA_MODE) {
 		nvdbg("WIFI DISCONNECTED AP - STA MODE");
 		strncpy(g_manager_info.ssid, "", 32);
-		strcpy(g_manager_info.ip4_address, "");
+		strncpy(g_manager_info.ip4_address, "", 18);
```
Comment (sunghan-chang) [inline framework/src/wifi_manager/wifi_manager.c]: I think below is enough to initialize, strncpy is too big functionality.
```
g_manager_info.ip4_address[0] = '\0';
```
How about change it?

File: framework/src/wifi_manager/wifi_manager.c [hunk 2]
```diff
@@ -490,7 +490,7 @@ wifi_manager_result_e wifi_manager_set_mode(wifi_manager_mode_e mode, wifi_manag
 		}
 
 		g_manager_info.mode = SOFTAP_MODE;
-		strncpy(g_manager_info.ssid, config->ssid, softap_config.ssid_length);
+		strncpy(g_manager_info.ssid, config->ssid, 32);
 		wifi_status_set(CLIENT_DISCONNECTED);
 		wifi_mutex_release(w_mutex);
```
Comment (girishks2000) [inline framework/src/wifi_manager/wifi_manager.c]: can you please state the issue with usage of strcpy

File: framework/src/wifi_manager/wifi_manager.c [hunk 3]
```diff
@@ -258,8 +258,8 @@ static wifi_manager_result_e wifi_fetch_scan_results(slsi_scan_info_t **wifi_sca
 		curr_record->rssi = wifi_scan_iter->rssi;
 		curr_record->channel = wifi_scan_iter->channel;
 		curr_record->phy_mode = wifi_scan_iter->phy_mode;
-		strncpy(curr_record->ssid, (char *)wifi_scan_iter->ssid, strlen((const char *)wifi_scan_iter->ssid));
-		strncpy(curr_record->bssid, (char *)wifi_scan_iter->bssid, strlen((const char *)wifi_scan_iter->bssid));
+		strncpy(curr_record->ssid, (char *)wifi_scan_iter->ssid, SLSI_SSID_LEN + 1);
```
Comment (sunghan-chang) [inline framework/src/wifi_manager/wifi_manager.c]: This module is the WiFi_manager, not slsi_wifi. Will you use SLSI_XX definition?

File: framework/src/wifi_manager/wifi_manager.c (hunk 1 above)
Comment (davidfather) [reply framework/src/wifi_manager/wifi_manager.c]: Done

File: framework/src/wifi_manager/wifi_manager.c (hunk 2 above)
Comment (davidfather) [reply framework/src/wifi_manager/wifi_manager.c]: strcpy is vulnerable. See the message below:
Use of vulnerable function 'strcpy' at wifi_manager.c. This function is unsafe, use strncpy instead.

File: framework/src/wifi_manager/wifi_manager.c (hunk 3 above)
Comment (davidfather) [reply framework/src/wifi_manager/wifi_manager.c]: current wifi_manager implementation is dependent on slsi_wifi.
I will consider an abstraction layer between wifi_manager and slsi_wifi when other wifi vendors join Tizen RT.

**Pull request merged**